### PR TITLE
Bump gcplogs and dependencies to v0.44.3

### PR DESCRIPTION
**Pull request by thaJeztah**
splitting this off of https://github.com/moby/moby/pull/39549 to see if we get the same oddness on Windows when using Go 1.12

- bump cloud.google.com/go v0.44.3 https://github.com/googleapis/google-cloud-go/compare/v0.23.0...v0.44.3
  ~- new (unused) dependency; `github.com/jstemmer/go-junit-report` encorced by cloud.google.com/go through https://github.com/googleapis/google-cloud-go/commit/c536dae6e0638a279604dde6dc84efe523b1a117#diff-39bad9c50a0f30bd6c067b83431bec76~
  ~- new (unused) dependency; `golang.org/x/tools` encorced by cloud.google.com/go through https://github.com/googleapis/google-cloud-go/commit/c536dae6e0638a279604dde6dc84efe523b1a117#diff-39bad9c50a0f30bd6c067b83431bec76~
- github.com/googleapis/gax-go https://github.com/googleapis/gax-go/compare/v2.0.0...v2.0.5
- go.opencensus.io https://github.com/census-instrumentation/opencensus-go/compare/v0.11.0...v0.22.3
- google.golang.org/api https://github.com/googleapis/google-api-go-client/compare/de943baf05a022a8f921b544b7827bacaba1aed5...v0.8.0

### CODE REVIEW

**Comment by thaJeztah** [comment]
Only failure on RS1 is; https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-39838/runs/1/nodes/39/log/?start=0

```
[2019-09-01T16:50:12.437Z] FAIL: docker_cli_ps_test.go:229: DockerSuite.TestPsListContainersFilterHealth
[2019-09-01T16:50:12.437Z] 
[2019-09-01T16:50:12.437Z] assertion failed: 
[2019-09-01T16:50:12.437Z] Command:  d:\CI-1\CI-a9a6716ca\binary\docker.exe run -d --name=none --no-healthcheck busybox sleep 240
[2019-09-01T16:50:12.437Z] ExitCode: 125
[2019-09-01T16:50:12.437Z] Error:    exit status 125
[2019-09-01T16:50:12.437Z] Stdout:   d286f429d6e91cdcb3a1144094a55c32a7a3ca5fec48b5b6db32a77adfa0b5bd
[2019-09-01T16:50:12.437Z] 
[2019-09-01T16:50:12.437Z] Stderr:   d:\CI-1\CI-a9a6716ca\binary\docker.exe: Error response from daemon: hcsshim::PrepareLayer - failed failed in Win32: This operation returned because the timeout period expired. (0x5b4).
[2019-09-01T16:50:12.437Z] 
[2019-09-01T16:50:12.437Z] 
[2019-09-01T16:50:12.437Z] Failures:
[2019-09-01T16:50:12.437Z] ExitCode was 125 expected 0
[2019-09-01T16:50:12.437Z] Expected no error
```

**Comment by thaJeztah** [comment]
@cpuguy83 @kolyshkin PTAL

**Comment by cpuguy83** [comment]
vendor conflicts.

**Comment by thaJeztah** [comment]
ah, protobuf was updated elsewhere I think

File: vendor.conf [hunk 1]
```diff
@@ -113,10 +113,13 @@ github.com/bsphere/le_go                            7a984a84b5492ae539b79b62fb4a
 
 # gcplogs deps
 golang.org/x/oauth2                                 bf48bf16ab8d622ce64ec6ce98d2c98f916b6303
-google.golang.org/api                               de943baf05a022a8f921b544b7827bacaba1aed5
-go.opencensus.io                                    c3ed530f775d85e577ca652cb052a52c078aad26 # v0.11.0
-cloud.google.com/go                                 0fd7230b2a7505833d5f69b75cbd6c9582401479 # v0.23.0
-github.com/googleapis/gax-go                        317e0006254c44a0ac427cc52a0e083ff0b9622f # v2.0.0
+google.golang.org/api                               02490b97dff7cfde1995bd77de808fd27053bc87 # v0.7.0
+github.com/golang/groupcache                        869f871628b6baa9cfbc11732cdf6546b17c1298
```
Comment (thaJeztah) [inline vendor.conf]: This replaced the dependency on `github.com/hashicorp/golang-lru`, which was Mozilla licensed and this one is Apache 2

**Comment by thaJeztah** [comment]
Hmmm..

```

[2020-03-06T23:35:21.436Z] # github.com/docker/docker/vendor/google.golang.org/grpc/balancer/grpclb
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb.go:61:3: unknown field 'MaxDelay' in struct literal of type "github.com/docker/docker/vendor/google.golang.org/grpc/internal/backoff".Exponential
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb.go:287:31: impossible type assertion:
[2020-03-06T23:35:21.436Z] 	*lbPicker does not implement balancer.Picker (wrong type for Pick method)
[2020-03-06T23:35:21.436Z] 		have Pick(context.Context, <T>) (balancer.SubConn, func(balancer.DoneInfo), error)
[2020-03-06T23:35:21.436Z] 		want Pick(context.Context, balancer.PickInfo) (balancer.SubConn, func(balancer.DoneInfo), error)
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_picker.go:99:52: undefined: balancer.PickOptions
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_picker.go:121:51: undefined: balancer.PickOptions
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_picker.go:157:51: undefined: balancer.PickOptions
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_remote_balancer.go:299:23: undefined: resolver.ResolveNowOption
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_remote_balancer.go:347:9: undefined: internal.WithResolverBuilder
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_util.go:70:79: undefined: resolver.BuildOption
[2020-03-06T23:35:21.436Z] vendor/google.golang.org/grpc/balancer/grpclb/grpclb_util.go:80:41: undefined: resolver.ResolveNowOption
```

**Comment by thaJeztah** [comment]
ah, more fun

```

[2020-04-02T22:21:05.205Z] ---> Making bundle: test-integration (in bundles/test-integration)
[2020-04-02T22:21:06.137Z] # github.com/docker/docker/integration-cli
[2020-04-02T22:21:06.137Z] import cycle not allowed
[2020-04-02T22:21:06.137Z] package github.com/docker/docker/integration-cli (test)
[2020-04-02T22:21:06.137Z] 	imports github.com/docker/docker/daemon/cluster/executor/container
[2020-04-02T22:21:06.137Z] 	imports github.com/docker/docker/daemon
[2020-04-02T22:21:06.137Z] 	imports github.com/docker/docker/daemon/logger/gcplogs
[2020-04-02T22:21:06.137Z] 	imports cloud.google.com/go/logging
[2020-04-02T22:21:06.137Z] 	imports cloud.google.com/go/logging/apiv2
[2020-04-02T22:21:06.137Z] 	imports github.com/googleapis/gax-go/v2
[2020-04-02T22:21:06.137Z] 	imports github.com/googleapis/gax-go/v2
```

**Review by cpuguy83 — APPROVED** [review]
+41k...

LGTM

**Comment by tiborvass** [comment]
@thaJeztah you don't need to add the unused dependencies, do you? It's under a `tools` tag. Vendor validation should pass. It would reduce the lines stats to `+15710, -5711`

**Comment by thaJeztah** [comment]
good point. I recall that things broke at the time somehow if I didn't include them, but it's been quite a while. Not at my computer, but feel free to push to my branch if you managed to get it working

**Review by tiborvass — APPROVED** [review]
`+17,150 −1,499`

**Comment by thaJeztah** [comment]
Thanks @tiborvass ❤️

**Comment by thaJeztah** [comment]
Ah, I just vendored a dependency, and I think this was the issue; perhaps we should exclude these in `hack/vendor.sh` to suppress the warnings;

```
2020/10/06 12:05:22 WARNING: dependency is not vendored: github.com/jstemmer/go-junit-report
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/cmd/goimports
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/internal/imports
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/go/ast/astutil
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/go/packages
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/internal/gopathwalk
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/internal/module
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/internal/semver
2020/10/06 12:05:22 WARNING: dependency is not vendored: golang.org/x/tools/internal/fastwalk
2020/10/06 12:05:23 WARNING: dependency is not vendored: golang.org/x/tools/go/gcexportdata
2020/10/06 12:05:23 WARNING: dependency is not vendored: golang.org/x/tools/go/internal/packagesdriver
2020/10/06 12:05:23 WARNING: dependency is not vendored: golang.org/x/tools/go/internal/gcimporter
2020/10/06 12:05:23 WARNING: dependency is not vendored: github.com/jstemmer/go-junit-report/formatter
2020/10/06 12:05:23 WARNING: dependency is not vendored: github.com/jstemmer/go-junit-report/parser
```

**Pull request merged**